presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2559,6 +2581,43 @@ private static HiveStorageFormat extractHiveStorageFormat(Table table)
         throw new PrestoException(HIVE_UNSUPPORTED_FORMAT, format("Output format %s with SerDe %s is not supported", outputFormat, serde));
     }
 
+    private static String encodePreferredOrderingColumns(List<SortingColumn> preferredOrderingColumns)
+    {
+        return Joiner.on(',').join(preferredOrderingColumns.stream()
+                .map(HiveMetadata::sortingColumnToString)
+                .collect(toImmutableList()));
+    }
+
+    private static List<SortingColumn> decodePreferredOrderingColumnsFromStorage(Storage storage)
+    {
+        if (!storage.getParameters().containsKey(PREFERRED_ORDERING_COLUMNS)) {
+            return ImmutableList.of();
+        }
+
+        return Splitter.on(',').trimResults().omitEmptyStrings().splitToList(storage.getParameters().get(PREFERRED_ORDERING_COLUMNS)).stream()
+                .map(HiveMetadata::sortingColumnFromString)
+                .collect(toImmutableList());
+    }
+
+    private static SortingColumn sortingColumnFromString(String name)
+    {
+        SortingColumn.Order order = ASCENDING;
+        String lower = name.toUpperCase(ENGLISH);
+        if (lower.endsWith(" ASC")) {
+            name = name.substring(0, name.length() - 4).trim();
```

Comment:
Why do we still need `trim`?

Reply:
These two methods are actually copied from `HiveTableProperties#sortingColumnFromString` and `HiveTableProperties#sortingColumnToString`. We could also make it a public static method so we don't need to copy around. What do you think?

Reply:
Per offline discussion, we moved these methods into `SortingColumn`.